### DISCUSSION TITLE
Data Files and input data

### DISCUSSION THREAD

**Discussion by furqan915** [neherlab/covid19_scenarios#872] [Q&A]
Hello,

I have just started working on COVID-19 Scenarios. I must admit this is a good algorithm and a web source. But there are certain queries that I need to ask.

In the data files of different states of USA, we have values for only 4 columns. But there are 6 columns and columns like hospitalized, icu, recoveries are not having any value.

> https://github.com/neherlab/covid19_scenarios/blob/master/data/case-counts/unitedstates/USA-Georgia.tsv

If I am going to upload my collected data, Do I need to follow the same pattern of the file mentioned? (No need to mention the Hospitalized, ICU and recovered data)

Thank you

**Comment by ivan-aksamentov** [comment]
Hi @furqan915 ,

In case counts data only `cases` and `time` columns are required. We describe file formats using [JSON schemas](https://json-schema.org/) in [`schemas/`](https://github.com/neherlab/covid19_scenarios/tree/110478ad73e416f99f8bfb95958b1ee5d0e767c5/schemas) directory .

In particular, see [`CaseCountsDatum.yml`](https://github.com/neherlab/covid19_scenarios/blob/110478ad73e416f99f8bfb95958b1ee5d0e767c5/schemas/CaseCountsDatum.yml).

Using those schemas, you can also generate types and validation functions for your favorite language.  For example, we use `quicktype` library to generate Python and Javascript/Typescript bindings [here](https://github.com/neherlab/covid19_scenarios/blob/110478ad73e416f99f8bfb95958b1ee5d0e767c5/tools/generateTypes.ts), and then we use the generated Python classes in `data/scripts` and Typescript types and runtime validators in `src/`.

The "Case counts" are the dots on the plot named "[Something] (data)" in the web app. The more columns you have the more useful the app is, as multiple curves will be then visible on the plots, and can be used to fit the parameters.

You might also find useful the code of our automatic data updater:
https://github.com/neherlab/covid19_scenarios/blob/110478ad73e416f99f8bfb95958b1ee5d0e767c5/.github/workflows/data-update.yml#L34-L48

**Reply by HaramFatima03** [reply]
Hello, 
May I please get to know, if this model is working on cumulative data for case counts file or its time based case-counts data for each of 6 categories?
